lib/OptimizationNLopt/src/OptimizationNLopt.jl

```diff
@@ -8,6 +8,7 @@ using Optimization.SciMLBase
 
 SciMLBase.allowsbounds(opt::Union{NLopt.Algorithm, NLopt.Opt}) = true
 SciMLBase.supports_opt_cache_interface(opt::Union{NLopt.Algorithm, NLopt.Opt}) = true
+SciMLBase.requiresgradient(opt::Union{NLopt.Algorithm, NLopt.Opt}) = true
```

Comment:
It would be useful to check if NLopt has a type hierarchy we can use here for doing this better, since it has all kinds of algorithms anything we put here will be wrong - in which case I would want to set the hessian to true as well.

Reply:
How about if I set the traits requiresgradient and requireshessian true for only the gradient-based algorithms (local and global). Only the derivative-free local optimizers will have these traits as false then.

Reply:
That's what I am hoping we can do yeah. But you should also be able to set the requireshessian to true only for second order methods by looking at the docs of nlopt. For the derivative-free vs others you plan to use the prefix in the names of the methods?